internal/bundler/linker.go

```diff
@@ -4320,10 +4347,25 @@ func (c *linkerContext) generateChunkJS(chunks []chunkInfo, chunkIndex int, chun
 	// deduplicated because some projects have thousands of files with the same
 	// comment. The comment must be preserved in the output for legal reasons but
 	// at the same time we want to generate a small bundle when minifying.
-	sort.Strings(commentList)
-	for _, text := range commentList {
-		j.AddString(text)
-		j.AddString("\n")
+	if len(legalCommentList) > 0 {
+		sort.Strings(legalCommentList)
+
+		switch c.options.LegalComments {
+		case config.LegalCommentsEndOfFile:
+			for _, text := range legalCommentList {
+				j.AddString(text)
+				j.AddString("\n")
```

Comment:
I think this line should come before `j.AddString(text)` so that users don't end up with an extra newline at the end of the file.

Reply:
Files generated by esbuild deliberately contain a trailing newline so that printing them to the terminal doesn't mess with the terminal prompt.